raco/algebra.py

```diff
@@ -344,6 +351,14 @@ def num_tuples(self):
         # TODO: use AGM bound (P10 in http://arxiv.org/pdf/1310.3314v2.pdf)
         return DEFAULT_CARDINALITY
 
+    def partitioning(self):
+        """attributes are mutually exclusive so union the partitioned
+        attributes"""
+        attributes = set()
+        for c in self.children():
+            attributes.union(c.partitioning().hash_partitioned)
```

Comment:
How can this work? `set.union(other)` doesn't mutate `self`, it just returns the union of `self` and `other`. Maybe `set.update()` is what you had in mind (I'd still rather avoid mutation for clarity)?